uni/lib/model/providers/startup/profile_provider.dart

```diff
@@ -38,17 +38,26 @@ class ProfileProvider extends StateProviderNotifier<Profile> {
   Future<Profile> loadFromRemote(StateProviders stateProviders) async {
     final session = stateProviders.sessionProvider.state!;
 
-    final profile = await fetchUserInfo(session);
+    final profileFuture = fetchUserInfo(session);
+    final courseUnitsFutures = profileFuture.then(
+      (profile) => fetchCourseUnitsAndCourseAverages(session, profile!),
+    );
 
-    final userBalanceAndFeesLimit = await fetchUserFeesBalanceAndLimit(session);
+    final futures = await Future.wait([
+      profileFuture,
+      fetchUserFeesBalanceAndLimit(session),
+      fetchUserPrintBalance(session),
+    ]);
```

Comment:
I think it might be more easy to understand if we move the `courseUnitsFuture` into here as well, extract it with the other ones and remove the await on line 60. What do you think?

Reply:
Agreed